WordPress/Classes/ViewRelated/System/Action Sheet/ActionSheetViewController.swift

```diff
@@ -0,0 +1,152 @@
+        return button
+    }()
+
+    @objc func buttonPressed() {
+        dismiss(animated: true, completion: nil)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.clipsToBounds = true
+        view.layer.cornerRadius = Constants.cornerRadius
+        view.layer.maskedCorners = [.layerMaxXMinYCorner, .layerMinXMinYCorner]
+        view.backgroundColor = .basicBackground
+
+        let headerLabelView = UIView()
+        let headerLabel = UILabel()
+        headerLabelView.addSubview(headerLabel)
+        headerLabelView.pinSubviewToAllEdges(headerLabel, insets: Constants.Header.insets)
+
+        headerLabel.font = WPStyleGuide.fontForTextStyle(.headline)
+        headerLabel.text = headerTitle
+        headerLabel.translatesAutoresizingMaskIntoConstraints = false
+
+        let buttonViews = buttons.map({ (buttonInfo) -> UIButton in
+            return button(buttonInfo)
+        })
+
+        NSLayoutConstraint.activate([
+            gripButton.heightAnchor.constraint(equalToConstant: Constants.gripHeight)
+        ])
+
+        let buttonConstraints = buttonViews.map { button in
+            return button.heightAnchor.constraint(equalToConstant: Constants.Button.height)
+        }
+
+        NSLayoutConstraint.activate(buttonConstraints)
+
+        let stackView = UIStackView(arrangedSubviews: [
+            gripButton,
+            headerLabelView
+        ] + buttonViews)
+
+        stackView.setCustomSpacing(Constants.Header.spacing, after: gripButton)
+        stackView.setCustomSpacing(Constants.Header.spacing, after: headerLabelView)
+
+        buttonViews.forEach { button in
+            stackView.setCustomSpacing(Constants.buttonSpacing, after: button)
+        }
+
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .vertical
+
+        refreshForTraits()
+
+        view.addSubview(stackView)
+        view.pinSubviewToSafeArea(stackView, insets: Constants.Stack.insets)
+    }
+
+    func button(_ info: ActionSheetButton) -> UIButton {
+        let button = UIButton(type: .custom)
+        button.setTitle(info.title, for: .normal)
+        button.titleLabel?.font = Constants.Button.font
+        button.setTitleColor(Constants.Button.textColor, for: .normal)
+        button.setImage(info.image, for: .normal)
+        button.imageView?.tintColor = Constants.Button.imageTintColor
+        button.setBackgroundImage(UIImage(color: .divider), for: .highlighted)
+        button.titleEdgeInsets = Constants.Button.titleInsets
+        button.naturalContentHorizontalAlignment = .leading
+        button.contentEdgeInsets = Constants.Button.contentInsets
+        button.addTarget(info.target, action: info.selector, for: .touchUpInside)
+        button.accessibilityIdentifier = info.identifier
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.flipInsetsForRightToLeftLayoutDirection()
```

Comment:
Yeah, I think this button should move to a subclass. Maybe like this?

```swift
extension UIButton {
    private enum Constants {
        static let contentInsets: UIEdgeInsets = UIEdgeInsets(top: 0, left: 18, bottom: 0, right: 35)
        static let titleInsets: UIEdgeInsets = UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 0)
        static let imageTintColor: UIColor = .neutral(.shade30)
        static let font: UIFont = .preferredFont(forTextStyle: .callout)
        static let textColor: UIColor = .text
    }
    
    convenience init(_ info: ActionSheetButton) {
        self.init(type: .custom)
        configureTitle(info.title)
        configureImage(info.image)
        configureInsets()
        setBackgroundImage(UIImage(color: .divider), for: .highlighted)
        addTarget(info.target, action: info.selector, for: .touchUpInside)
        accessibilityIdentifier = info.identifier
    }
        
    private func configureTitle(_ titleString: String) {
        setTitle(titleString, for: .normal)
        titleLabel?.font = Constants.font
        setTitleColor(Constants.textColor, for: .normal)
    }
    
    private func configureImage(_ image: UIImage) {
        setImage(image, for: .normal)
        imageView?.tintColor = Constants.imageTintColor
    }
    
    private func configureInsets() {
        contentEdgeInsets = Constants.contentInsets
        titleEdgeInsets = Constants.titleInsets
        naturalContentHorizontalAlignment = .leading
        flipInsetsForRightToLeftLayoutDirection()
    }
}
```

I think it might be best to keep that change + the Stack View subclass as a separate PR.

Reply:
That works for me!

Reply:
This looks like a good improvement :)